File: morning_exercises/day5.cs
public static class CheckValid
{
    public static bool isValid(string input)
    {
        var stack = new Stack<char>();
        var mappings = new Dictionary<char, char>
        {
            ['}'] = '{',
            [')'] = '(',
            [']'] = '['
        };

        foreach (char c in input)
        {
            if (c=='{' || c=='[' || c=='(') stack.Push(c);
            else
            {
                if (stack.Count==0) return false;

                if(mappings[c]!=stack.Peek()) return false;
                stack.Pop();
            }
        }
        return stack.Count==0;
    }
}

File: morning_exercises/day5.cpp
#include <stack>
#include <string>
#include <iostream>

bool isValid(std::string input) {
    std::stack<char> openings;

    for (int i=0; i<input.length(); i++) {
        if (input[i]=='(' || input[i]=='{' || input[i]=='[') openings.push(input[i]);
        else {
            if (openings.empty()) return false;

            if (input[i]==')' && openings.top()=='(') openings.pop();
            else if (input[i]=='}' && openings.top()=='{') openings.pop();
            else if (input[i]==']' && openings.top()=='[') openings.pop();
            else return false;
        }
    }
    return openings.empty();
}

int main() {
    std::cout << std::boolalpha;
    std::cout << isValid("{{]}}") << "\n";
    std::cout << isValid("()[]{}") << "\n";
    std::cout << isValid("{[]}") << "\n";
}